pkg/controller/statefulset/stateful_set_test.go

```diff
@@ -534,6 +539,34 @@ func TestGetPodsForStatefulSetAdopt(t *testing.T) {
 	}
 }
 
+func TestAdoptOrphanRevisions(t *testing.T) {
+	ss1 := newStatefulSetWithLabels(3, "ss1", types.UID("ss1"), map[string]string{"foo": "bar"})
+	ss1.Status.CollisionCount = new(int32)
+	ss1Rev1, err := history.NewControllerRevision(ss1, parentKind, ss1.Spec.Template.Labels, rawTemplate(&ss1.Spec.Template), 1, ss1.Status.CollisionCount)
+	if err != nil {
+		t.Fatal(err)
+	}
+	ss1Rev1.Namespace = ss1.Namespace
+	ss1.Spec.Template.Annotations = make(map[string]string)
+	ss1.Spec.Template.Annotations["ss1"] = "ss1"
+	ss1Rev2, err := history.NewControllerRevision(ss1, parentKind, ss1.Spec.Template.Labels, rawTemplate(&ss1.Spec.Template), 2, ss1.Status.CollisionCount)
+	if err != nil {
+		t.Fatal(err)
+	}
+	ss1Rev2.Namespace = ss1.Namespace
+	ss1Rev2.OwnerReferences = []metav1.OwnerReference{}
+
+	ssc, spc := newFakeStatefulSetController(ss1, ss1Rev1, ss1Rev2)
+
+	spc.revisionsIndexer.Add(ss1Rev1)
+	spc.revisionsIndexer.Add(ss1Rev2)
+
+	err = ssc.adoptOrphanRevisions(ss1)
+	if err != nil {
+		t.Errorf("adoptOrphanRevisions() error: %v", err)
+	}
```

Comment:
now we should verify, the revision got adopted, which means it has owner ref for ss1Rev2 ?

Reply:
Yes, added the logic, thanks